From b45f5dbf100dff914c8f729e049ecaf26073030e Mon Sep 17 00:00:00 2001
From: Lu Jingxiao <lujingxiao@huawei.com>
Date: Tue, 4 Jan 2022 15:55:35 +0800
Subject: [PATCH 01/20] isula-build: update go version in go.mod and README to
 1.15

After upgrade vendor, the required go version is 1.15 or later.

Fixes: #I4KE3A

Signed-off-by: Lu Jingxiao <lujingxiao@huawei.com>
---
 README.md          |  2 +-
 README.zh.md       |  2 +-
 go.mod             |  2 +-
 vendor/modules.txt | 33 +++++++++++++++++++++++++++++++++
 4 files changed, 36 insertions(+), 3 deletions(-)

diff --git a/README.md b/README.md
index 8c09b1b..a864363 100644
--- a/README.md
+++ b/README.md
@@ -39,7 +39,7 @@ In addition, the following capabilities are also provided:
 For compiling from source on openEuler, these packages are required on your OS:
 
 - make
-- golang (version 1.13 or later)
+- golang (version 1.15 or later)
 - btrfs-progs-devel
 - device-mapper-devel
 - glib2-devel
diff --git a/README.zh.md b/README.zh.md
index 15301c0..e9c2747 100644
--- a/README.zh.md
+++ b/README.zh.md
@@ -41,7 +41,7 @@ isula-build提供的命令行工具提供了很多功能，例如：
 为了顺利从源码编译，以下包需要被安装在你的操作系统中：
 
 - make
-- golang（大于等于1.13版本）
+- golang（大于等于1.15版本）
 - btrfs-progs-devel
 - device-mapper-devel
 - glib2-devel
diff --git a/go.mod b/go.mod
index f60eeea..dbf9426 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module isula.org/isula-build
 
-go 1.13
+go 1.15
 
 require (
 	github.com/BurntSushi/toml v0.4.1
diff --git a/vendor/modules.txt b/vendor/modules.txt
index 660de5e..ec0b737 100644
--- a/vendor/modules.txt
+++ b/vendor/modules.txt
@@ -1,4 +1,5 @@
 # github.com/BurntSushi/toml v0.4.1
+## explicit
 github.com/BurntSushi/toml
 github.com/BurntSushi/toml/internal
 # github.com/Microsoft/go-winio v0.5.0
@@ -35,8 +36,10 @@ github.com/acarl005/stripansi
 # github.com/beorn7/perks v1.0.1
 github.com/beorn7/perks/quantile
 # github.com/blang/semver v4.0.0+incompatible
+## explicit
 github.com/blang/semver
 # github.com/bndr/gotabulate v1.1.3-0.20210209140214-21a495b00e22
+## explicit
 github.com/bndr/gotabulate
 # github.com/cespare/xxhash/v2 v2.1.1
 github.com/cespare/xxhash/v2
@@ -45,6 +48,7 @@ github.com/containerd/cgroups/stats/v1
 # github.com/containerd/console v1.0.2
 github.com/containerd/console
 # github.com/containerd/containerd v1.5.8
+## explicit
 github.com/containerd/containerd/errdefs
 github.com/containerd/containerd/log
 github.com/containerd/containerd/pkg/userns
@@ -52,11 +56,13 @@ github.com/containerd/containerd/platforms
 github.com/containerd/containerd/sys
 github.com/containerd/containerd/sys/reaper
 # github.com/containerd/go-runc v1.0.0
+## explicit
 github.com/containerd/go-runc
 # github.com/containerd/stargz-snapshotter/estargz v0.9.0
 github.com/containerd/stargz-snapshotter/estargz
 github.com/containerd/stargz-snapshotter/estargz/errorutil
 # github.com/containers/image/v5 v5.17.0
+## explicit
 github.com/containers/image/v5/copy
 github.com/containers/image/v5/directory
 github.com/containers/image/v5/directory/explicitfilepath
@@ -104,6 +110,7 @@ github.com/containers/image/v5/transports/alltransports
 github.com/containers/image/v5/types
 github.com/containers/image/v5/version
 # github.com/containers/libtrust v0.0.0-20200511145503-9c3a6c22cd9a
+## explicit
 github.com/containers/libtrust
 # github.com/containers/ocicrypt v1.1.2
 github.com/containers/ocicrypt
@@ -121,6 +128,7 @@ github.com/containers/ocicrypt/spec
 github.com/containers/ocicrypt/utils
 github.com/containers/ocicrypt/utils/keyprovider
 # github.com/containers/storage v1.37.0
+## explicit
 github.com/containers/storage
 github.com/containers/storage/drivers
 github.com/containers/storage/drivers/aufs
@@ -166,10 +174,13 @@ github.com/containers/storage/pkg/truncindex
 github.com/containers/storage/pkg/unshare
 github.com/containers/storage/types
 # github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
+## explicit
 github.com/coreos/go-systemd/daemon
 # github.com/cyphar/filepath-securejoin v0.2.3
+## explicit
 github.com/cyphar/filepath-securejoin
 # github.com/docker/distribution v2.7.1+incompatible
+## explicit
 github.com/docker/distribution
 github.com/docker/distribution/digestset
 github.com/docker/distribution/metrics
@@ -182,6 +193,7 @@ github.com/docker/distribution/registry/client/transport
 github.com/docker/distribution/registry/storage/cache
 github.com/docker/distribution/registry/storage/cache/memory
 # github.com/docker/docker v20.10.11+incompatible
+## explicit
 github.com/docker/docker/api
 github.com/docker/docker/api/types
 github.com/docker/docker/api/types/blkiodev
@@ -215,14 +227,17 @@ github.com/docker/go-metrics
 # github.com/docker/go-units v0.4.0
 github.com/docker/go-units
 # github.com/docker/libnetwork v0.8.0-dev.2.0.20190625141545-5a177b73e316
+## explicit
 github.com/docker/libnetwork/resolvconf
 github.com/docker/libnetwork/resolvconf/dns
 github.com/docker/libnetwork/types
 # github.com/ghodss/yaml v1.0.0
 github.com/ghodss/yaml
 # github.com/gofrs/flock v0.8.1
+## explicit
 github.com/gofrs/flock
 # github.com/gogo/protobuf v1.3.2
+## explicit
 github.com/gogo/protobuf/gogoproto
 github.com/gogo/protobuf/proto
 github.com/gogo/protobuf/protoc-gen-gogo/descriptor
@@ -257,6 +272,7 @@ github.com/imdario/mergo
 # github.com/inconshreveable/mousetrap v1.0.0
 github.com/inconshreveable/mousetrap
 # github.com/ishidawataru/sctp v0.0.0-20191218070446-00ab2ac2db07
+## explicit
 github.com/ishidawataru/sctp
 # github.com/json-iterator/go v1.1.12
 github.com/json-iterator/go
@@ -282,6 +298,8 @@ github.com/miekg/pkcs11
 github.com/mistifyio/go-zfs
 # github.com/moby/sys/mountinfo v0.4.1
 github.com/moby/sys/mountinfo
+# github.com/moby/term v0.0.0-20201216013528-df9cb8a40635
+## explicit
 # github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
 github.com/modern-go/concurrent
 # github.com/modern-go/reflect2 v1.0.2
@@ -289,16 +307,20 @@ github.com/modern-go/reflect2
 # github.com/mtrmac/gpgme v0.1.2
 github.com/mtrmac/gpgme
 # github.com/opencontainers/go-digest v1.0.0
+## explicit
 github.com/opencontainers/go-digest
 # github.com/opencontainers/image-spec v1.0.2-0.20210819154149-5ad6f50d6283
+## explicit
 github.com/opencontainers/image-spec/specs-go
 github.com/opencontainers/image-spec/specs-go/v1
 # github.com/opencontainers/runc v1.0.2
 github.com/opencontainers/runc/libcontainer/user
 github.com/opencontainers/runc/libcontainer/userns
 # github.com/opencontainers/runtime-spec v1.0.3-0.20210326190908-1c3f411f0417
+## explicit
 github.com/opencontainers/runtime-spec/specs-go
 # github.com/opencontainers/runtime-tools v0.9.0
+## explicit
 github.com/opencontainers/runtime-tools/error
 github.com/opencontainers/runtime-tools/filepath
 github.com/opencontainers/runtime-tools/generate
@@ -314,6 +336,7 @@ github.com/opencontainers/selinux/pkg/pwalkdir
 github.com/ostreedev/ostree-go/pkg/glibobject
 github.com/ostreedev/ostree-go/pkg/otbuiltin
 # github.com/pkg/errors v0.9.1
+## explicit
 github.com/pkg/errors
 # github.com/prometheus/client_golang v1.7.1
 github.com/prometheus/client_golang/prometheus
@@ -322,6 +345,7 @@ github.com/prometheus/client_golang/prometheus/promhttp
 # github.com/prometheus/client_model v0.2.0
 github.com/prometheus/client_model/go
 # github.com/prometheus/common v0.11.1
+## explicit
 github.com/prometheus/common/expfmt
 github.com/prometheus/common/internal/bitbucket.org/ww/goautoneg
 github.com/prometheus/common/model
@@ -332,14 +356,17 @@ github.com/prometheus/procfs/internal/util
 # github.com/rivo/uniseg v0.2.0
 github.com/rivo/uniseg
 # github.com/sirupsen/logrus v1.8.1
+## explicit
 github.com/sirupsen/logrus
 # github.com/spf13/cobra v1.2.1
+## explicit
 github.com/spf13/cobra
 # github.com/spf13/pflag v1.0.5
 github.com/spf13/pflag
 # github.com/stefanberger/go-pkcs11uri v0.0.0-20201008174630-78d3cae3a980
 github.com/stefanberger/go-pkcs11uri
 # github.com/syndtr/gocapability v0.0.0-20200815063812-42c35b437635
+## explicit
 github.com/syndtr/gocapability/capability
 # github.com/tchap/go-patricia v2.3.0+incompatible
 github.com/tchap/go-patricia/patricia
@@ -358,6 +385,7 @@ github.com/vbauerster/mpb/v7/cwriter
 github.com/vbauerster/mpb/v7/decor
 github.com/vbauerster/mpb/v7/internal
 # github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb
+## explicit
 github.com/xeipuuv/gojsonpointer
 # github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
 github.com/xeipuuv/gojsonreference
@@ -374,6 +402,7 @@ go.opencensus.io/trace
 go.opencensus.io/trace/internal
 go.opencensus.io/trace/tracestate
 # golang.org/x/crypto v0.0.0-20211117183948-ae814b36b871
+## explicit
 golang.org/x/crypto/cast5
 golang.org/x/crypto/ed25519
 golang.org/x/crypto/ed25519/internal/edwards25519
@@ -396,9 +425,11 @@ golang.org/x/net/internal/timeseries
 golang.org/x/net/proxy
 golang.org/x/net/trace
 # golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
+## explicit
 golang.org/x/sync/errgroup
 golang.org/x/sync/semaphore
 # golang.org/x/sys v0.0.0-20211124211545-fe61309f8881
+## explicit
 golang.org/x/sys/execabs
 golang.org/x/sys/internal/unsafeheader
 golang.org/x/sys/plan9
@@ -414,6 +445,7 @@ golang.org/x/text/unicode/norm
 # google.golang.org/genproto v0.0.0-20211005153810-c76a74d43a8e
 google.golang.org/genproto/googleapis/rpc/status
 # google.golang.org/grpc v1.42.0
+## explicit
 google.golang.org/grpc
 google.golang.org/grpc/attributes
 google.golang.org/grpc/backoff
@@ -497,6 +529,7 @@ gopkg.in/square/go-jose.v2/json
 # gopkg.in/yaml.v2 v2.4.0
 gopkg.in/yaml.v2
 # gotest.tools/v3 v3.0.3
+## explicit
 gotest.tools/v3/assert
 gotest.tools/v3/assert/cmp
 gotest.tools/v3/fs
-- 
2.27.0

